src/compiler/commandLineParser.ts

```diff
@@ -2913,6 +2917,11 @@ namespace ts {
             }
         }
 
+        // eslint-disable-next-line no-in-operator
+        if (rootCompilerOptions && json && !("compilerOptions" in json)) {

```

Comment:
if you just want to do it now

```suggestion
        if (rootCompilerOptions && json && json.compilerOptions === undefined) {
```